pygmt/src/grdfilter.py

```diff
@@ -48,64 +48,59 @@ def grdfilter(grid, **kwargs):
-
-        g: Gaussian
-
-        o: Operator
-
-        m: Median
-
-        p: Maximum Likelihood probability
-
-        h: histogram
     distance : str
-        Distance *flag* tells how grid (x,y) relates to filter width as
-        follows:
+        State how the grid (x,y) relates to filter *width*:
 
-        p: grid (px,py) with *width* an odd number of pixels; Cartesian
-        distances.
-
-        0: grid (x,y) same units as *width*, Cartesian distances.
-
-        1: grid (x,y) in degrees, *width* in kilometers, Cartesian distances.
-
-        2: grid (x,y) in degrees, *width* in km, dx scaled by cos(middle y),
-        Cartesian distances.
+        - ``"p"``: grid (px,py) with *width* an odd number of pixels,
+          Cartesian distances.
+        - ``"0"``: grid (x,y) same units as *width*, Cartesian distances.
+        - ``"1"``: grid (x,y) in degrees, *width* in kilometers, Cartesian
+          distances.
+        - ``"2"``: grid (x,y) in degrees, *width* in km, dx scaled by
+          cos(middle y), Cartesian distances.
 
         The above options are fastest because they allow weight matrix to be
         computed only once. The next three options are slower because they
         recompute weights for each latitude.
 
-        3: grid (x,y) in degrees, *width* in km, dx scaled by cosine(y),
-        Cartesian distance calculation.
-
-        4: grid (x,y) in degrees, *width* in km, Spherical distance
-        calculation.
-
-        5: grid (x,y) in Mercator ``projection="m1"`` img units, *width* in km,
-        Spherical distance calculation.
+        - ``"3"``: grid (x,y) in degrees, *width* in km, dx scaled by cos(y),
+          Cartesian distance calculation.
+        - ``"4"``: grid (x,y) in degrees, *width* in km, Spherical distance
+          calculation.
+        - ``"5"``: grid (x,y) in Mercator ``projection="m1"`` img units,
+          *width* in km, Spherical distance calculation.
 
     {spacing}
     nans : str or float
         **i**\|\ **p**\|\ **r**.
-        Determine how NaN-values in the input grid affects the filtered output.
+        Determine how NaN-values in the input grid affect the filtered output.
+        Use **i** to ignore all NaNs in the calculation of the filtered value
+        [Default]. **r** is same as **i** except if the input node was NaN then
+        the output node will be set to NaN (only applies if both grids are
+        co-registered). **p** will force the filtered value to be NaN if any
+        grid nodes with NaN-values are found inside the filter circle.
     {region}
     toggle : bool
-        Toggle the node registration for the output grid so as to become the
-        opposite of the input grid. [Default gives the same registration as the
-        input grid].
+        Toggle the node registration for the output grid to get the opposite of
+        the input grid [Default gives the same registration as the input grid].
     {verbose}
     {coltypes}
     {registration}
```

Comment:
Not documented upstream yet, but I've tested `grdfilter -x` at https://forum.generic-mapping-tools.org/t/pygmt-grdfilter-calculations-will-be-distributed-over-1-threads/4147/8 and it works. Will need to add `x="cores"` at L28 also.

```suggestion
    {registration}
    {cores}
```

Reply:
Should I submit a separate PR for adding this alias, or can this be included in this PR as the docstring for `cores` is already in `decorators.py`?
We probably also need an upstream PR to update the GMT documentation.

Reply:
> Should I submit a separate PR for adding this alias, or can this be included in this PR as the docstring for `cores` is already in `decorators.py`?

Ok to do it here since it's just a 2 line change, and all the other aliases for grdfilter are wrapped already so this will complete things.

> We probably also need an upstream PR to update the GMT documentation.

Yes please, if you have time!

Reply:
> Will need to add x="cores" at L28 also.

Done in commit https://github.com/GenericMappingTools/pygmt/pull/2617/commits/e5ef014a2754130960f5b562cbe2ae6b27face26.

Reply:
Awesome, thanks!